### PR TITLE
feat(KNO-4647) Update management api reference

### PR DESCRIPTION
**Pull request by francoborr**
### Description
Adds the following sections to the Management Api reference:
* Commit description
* Commits get endpoint
* Commits list endpoint
* Commits promote one endpoint

### Tasks
[KNO-4647](https://linear.app/knock/issue/KNO-4647/[docs]-update-management-api-reference)

### Screenshots

https://github.com/knocklabs/docs/assets/67347884/d6a72d19-7ed2-4d6e-b263-9d31c4dcb780

### CODE REVIEW

File: content/mapi.mdx [hunk 1]
```diff
@@ -27,13 +27,14 @@ tags: ["api", "sdks", "api key", "keys", "errors"]
...
-The Knock management API provides you with a programmatic way to interact with the resources you create and manage in your Knock dashboard, including workflows, templates, layouts, and translations. It's separate from the [Knock API](/reference) and only provides access to a limited subset of resources.
+The Knock management API provides you with a programmatic way to interact with the resources you create and manage in your Knock dashboard, including workflows, templates, layouts, translations and commits. It's separate from the [Knock API](/reference) and only provides access to a limited subset of resources.
 
 You can use the Knock management API to:
 
 - Create, update, and manage your Knock workflows and the notification templates within those workflows.
 - Create and manage the [translations](/send-and-manage-data/translations) used by your notification templates.
-- Commit and promote changes between your Knock environments.
+- Create, update and manage your [email layouts](/integrations/email/layouts).
+- Commit and promote changes between your Knock environments, and list commits in a given environment.
```
Comment (thomaswhyyou) [inline content/mapi.mdx:37]:
```suggestion
- Commit and promote changes between your Knock environments.
```
I personally think this bullet was fine as it was. We don't necessarily spell out every command for other resources either.

File: content/mapi.mdx [hunk 2]
```diff
@@ -2067,3 +2303,49 @@ Promote all changes across all resources to the target environment from its prec
...
+      "id": "e4a22631-e05e-4d40-b323-4dc327c0670e",
+      "resource": {
+         "identifier": "new-comment",
+         "type": "workflow"
+      },
+      "author": {
+         "email": "franklin@gmail.com",
+         "name": "Franklin Sierra"
+      },
+      "environment": "development"
```
Comment (thomaswhyyou) [inline content/mapi.mdx:2345]:
```suggestion
      "environment": "staging"
```
I think it's a bit weird to show a commit in development environment as an example payload of a promoted commit.

File: content/mapi.mdx [hunk 3]
```diff
@@ -1975,22 +1976,257 @@ A translation object
...
+  />
+</Attributes>
+
+### CommitResource Attributes
+
+<Attributes>
+    <Attribute
+    name="identifier"
+    type="string"
+    description="The Unique identifier for the resource."
```
Comment (thomaswhyyou) [inline content/mapi.mdx:2049]:
```suggestion
    description="The unique identifier for the resource."
```

File: content/mapi.mdx [hunk 4]
```diff
@@ -1975,22 +1976,257 @@ A translation object
...
+  />
+    <Attribute
+    name="type"
+    type="string"
+    description="The type of the resource object."
+  />
+</Attributes>
+
+### CommitAuthor Attributes
```
Comment (thomaswhyyou) [inline content/mapi.mdx:2058]:
```suggestion
### CommitAuthor attributes
```

File: content/mapi.mdx [hunk 5]
```diff
@@ -1975,22 +1976,257 @@ A translation object
...
+    description="The environment where the commit was made."
+  />
+  <Attribute
+    name="created_at"
+    type="timestamp"
+    description="A timestamp of when the commit was created."
+  />
+</Attributes>
+
+### CommitResource Attributes
```
Comment (thomaswhyyou) [inline content/mapi.mdx:2043]:
```suggestion
### CommitResource attributes
```

File: content/mapi.mdx [hunk 6]
```diff
@@ -1975,22 +1976,257 @@ A translation object
...
+  />
+</Attributes>
+
+### CommitAuthor Attributes
+
+<Attributes>
+    <Attribute
+    name="email"
+    type="string"
+    description="The email address of the author."
```
Comment (thomaswhyyou) [inline content/mapi.mdx:2064]:
```suggestion
    description="The email address of the commit author."
```

File: content/mapi.mdx [hunk 7]
```diff
@@ -1975,22 +1976,257 @@ A translation object
...
+### Query parameters
+
+<Attributes>
+  <Attribute
+    name="environment"
+    type="string"
+    description="A slug of the environment from which to query commits."
+  />  
+  <Attribute
+    name="[no-]promoted"
```
Comment (thomaswhyyou) [inline content/mapi.mdx:2114]:
```suggestion
    name="promoted"
```
I think this should just be a `promoted` param?

File: content/mapi.mdx [hunk 8]
```diff
@@ -1975,22 +1976,257 @@ A translation object
...
+    type="string"
```
Comment (thomaswhyyou) [inline content/mapi.mdx:2115]:
```suggestion
    type="boolean"
```
I believe this is a boolean param.

**Review by thomaswhyyou — COMMENTED** [review]
A few small things! Thanks for preparing this PR 🙏

File: content/mapi.mdx (hunk 7 above)
Comment (francoborr) [reply content/mapi.mdx:2114]: I wanted to make clear that it can be `--promoted` or `--no-promoted`, similar to what we show on the CLI:

Comment (thomaswhyyou) [reply content/mapi.mdx:2114]: But that's the CLI option though, not a param for the endpoint? There is no `no-promoted` param for this endpoint:

https://github.com/knocklabs/control/blob/57d8f14e2d340c7475504db6494364188b4db9a1/backend/lib/control_web/controllers/v1/schemas/commit_schema.ex#L11

Comment (francoborr) [reply content/mapi.mdx:2114]: Oh you are right, I got confused. Thanks for pointing this!

**Review by thomaswhyyou — APPROVED** [review]

**Pull request merged**